cygwin/CygwinProcessTable.c

```diff
@@ -0,0 +1,799 @@
+   char* location = strchr(buf, ' ');
+   if (!location)
+      return false;
+
+   /* (2) comm  -  (%s) */
+   location += 2;
+   char* end = strrchr(location, ')');
+   if (!end)
+      return false;
+
+   String_safeStrncpy(command, location, MINIMUM((size_t)(end - location + 1), commLen));
+
+   location = end + 2;
+
+   /* (3) state  -  %c */
+   process->state = CygwinProcessTable_getProcessState(location[0]);
+   location += 2;
+
+   /* (4) ppid  -  %d */
+   Process_setParent(process, strtol(location, &location, 10));
+   location += 1;
+
+   /* (5) pgrp  -  %d */
+   process->pgrp = strtol(location, &location, 10);
+   location += 1;
+
+   /* (6) session  -  %d */
+   process->session = strtol(location, &location, 10);
+   location += 1;
+
+   /* (7) tty_nr  -  %d */
+   process->tty_nr = strtoul(location, &location, 10);
+   location += 1;
+
+   /* (8) tpgid  -  %d */
+   process->tpgid = strtol(location, &location, 10);
+   location += 1;
+
+   /* (9) flags  -  %u */
+   cp->flags = strtoul(location, &location, 10);
+   location += 1;
+
+   /* (10) minflt  -  %lu */
+   process->minflt = strtoull(location, &location, 10);
+   location += 1;
+
+   /* (11) cminflt  -  %lu */
+   cp->cminflt = strtoull(location, &location, 10);
+   location += 1;
+
+   /* (12) majflt  -  %u */
+   process->majflt = strtoul(location, &location, 10);
+   location += 1;
+
+   /* (13) cmajflt  -  %u */
+   cp->cmajflt = strtoul(location, &location, 10);
+   location += 1;
+
+   /* (14) utime  -  %lu */
+   cp->utime = CygwinProcessTable_adjustTime(chost, strtoull(location, &location, 10));
+   location += 1;
+
+   /* (15) stime  -  %lu */
+   cp->stime = CygwinProcessTable_adjustTime(chost, strtoull(location, &location, 10));
+   location += 1;
+
+   /* (16) cutime  -  %lu */
+   cp->cutime = CygwinProcessTable_adjustTime(chost, strtoull(location, &location, 10));
+   location += 1;
+
+   /* (17) cstime  -  %lu */
+   cp->cstime = CygwinProcessTable_adjustTime(chost, strtoull(location, &location, 10));
+   location += 1;
+
+   /* (18) priority  -  %d */
+   process->priority = strtol(location, &location, 10);
+   location += 1;
+
+   /* (19) nice  -  %d */
+   process->nice = strtol(location, &location, 10);
+   location += 1;
+
+   /* (20) num_threads  -  %d */
+   process->nlwp = strtol(location, &location, 10);
+   location += 1;
+
+   /* Skip (21) itrealvalue  -  %d */
+   location = strchr(location, ' ') + 1;
+
+   /* (22) starttime  -  %lu */
+   if (process->starttime_ctime == 0) {
+      process->starttime_ctime = chost->boottime + CygwinProcessTable_adjustTime(chost, strtoull(location, &location, 10)) / 100;
+   } else {
+      location = strchr(location, ' ');
+   }
+
+   /* Ignore further fields */
+
+   process->processor = 0;  // FIXME
+
+   process->time = cp->utime + cp->stime;
+
+   return true;
+}
+
+static bool CygwinProcessTable_updateUser(const Machine* host, Process* process, openat_arg_t procFd) {
```

Comment:
Should be made SID-aware

Reply:
> Should be made SID-aware

it's session ID, not security identifier...

see:
https://github.com/cygwin/cygwin/blob/main/winsup/cygwin/fhandler/process.cc#L1197
https://github.com/cygwin/cygwin/blob/a9e8e3d1cb8235f513f4d8434509acf287494fcf/winsup/cygwin/local_includes/pinfo.h#L87